src/main/java/org/broadinstitute/hellbender/utils/GenotypeUtils.java

```diff
@@ -125,4 +139,50 @@ public static GenotypeCounts computeDiploidGenotypeCounts(final VariantContext v
     public static boolean genotypeIsUsableForAFCalculation(Genotype g) {
         return g.hasLikelihoods() || g.hasGQ() || g.getAlleles().stream().anyMatch(a -> a.isCalled() && a.isNonReference() && !a.isSymbolic());
     }
+
+    /**
+     * For a hom-ref, as long as we have GQ we can make a very accurate QUAL calculation
+     * since the hom-var likelihood should make a minuscule contribution
+     * @param g
+     * @param nAlleles
+     * @return log10 likelihoods
+     */
+    public static double[] makeApproximateLog10LikelihoodsFromGQ(Genotype g, int nAlleles) {
+        final int[] perSampleIndexesOfRelevantAlleles = new int[nAlleles];
+        Arrays.fill(perSampleIndexesOfRelevantAlleles, 1);
+        perSampleIndexesOfRelevantAlleles[0] = 0;  //ref still maps to ref
+        //use these values for diploid ref/ref, ref/alt, alt/alt likelihoods
+        final int gq = g.getGQ();
+        final int ploidy = g.getPloidy();
+        final int[] approxLikelihoods = {0, gq, PLOIDY_2_HOM_VAR_SCALE_FACTOR*gq};
+        //map likelihoods for any other alts to biallelic ref/alt likelihoods above
+        final int[] genotypeIndexMapByPloidy = GL_CALCS.getInstance(ploidy, nAlleles).genotypeIndexMap(perSampleIndexesOfRelevantAlleles, GL_CALCS); //probably horribly slow
+        final int[] PLs = new int[genotypeIndexMapByPloidy.length];
+        for (int i = 0; i < PLs.length; i++) {
+            PLs[i] = approxLikelihoods[genotypeIndexMapByPloidy[i]];
+        }
+        return GenotypeLikelihoods.fromPLs(PLs).getAsVector();  //fromPLs converts from Phred-space back to log10-space
+    }
+
+    public static boolean hasData(final Genotype g) {
+        if (g.hasPL() && !g.isNonInformative()) {
+            return true;
+        }
+        else if (g.hasGQ() && g.getGQ() > 0) {
+            return true;
+        } else {
+            return false;
+        }
+    }
+
+    public static boolean shouldBeCalled(final Genotype g) {
+        if (g.hasPL() && !g.isNonInformative()) {
```

Comment:
Similarly:
```
return !g.isNonInformatiive() || g.hasGQ();
```

Now in insolation (not looking how this is used) if the only think it have is a GQ=0, how is that callable?

Reply:
I agree that GQ0 shouldn't be called, but right now I'm just matching the existing behavior.  I will open an issue to fix the GQ0 case.